neo/Consensus/ConsensusService.cs

```diff
@@ -193,8 +187,9 @@ private void OnChangeViewReceived(ConsensusPayload payload, ChangeView message)
                 if (!context.CommitSent())
                 {
                     bool shouldSendRecovery = false;
-                    // Limit recovery to sending from `f` nodes when the request is from a lower view number.
-                    int allowedRecoveryNodeCount = context.F();
+                    // Limit recovery to be sent from, at least, `f+1` nodes when the request is from a lower view number.
+                    // However, `f+2` was chosen for ensuring a possible faster reply for those asking recovery
+                    int allowedRecoveryNodeCount = context.F() + 2;
```

Comment:
I think `F` nodes sending `RecoveryMessage` is enough. Because `payload.ValidatorIndex` is one of the fault node, so there is at least one good node in another `F` nodes.

Reply:
This morning we got the following error:
We had 4 nodes.
Index 3 was the failed one.


                    for (int i = 0; i < allowedRecoveryNodeCount; i++)
                    {                          
                        var eligibleResponders = context.Validators.Length - 1; // 3
                        // (1 + 0 + 1) % 3 = 2
                        var chosenIndex = (payload.ValidatorIndex + i + message.NewViewNumber) % eligibleResponders;
                        // 2 >= 1 => chosenIndex++ = 3
                        if (chosenIndex >= payload.ValidatorIndex) chosenIndex++;
                        // however index 3 was intentionally dead!
                        if (chosenIndex != context.MyIndex) continue;
                        shouldSendRecovery = true;
                        break;
                    }

Maybe with the next fix on NewViewNumber and its static setting to ViewNumber++ we gonna be able to recover with `F`. Let's analyse this.

Reply:
> We had 4 nodes.
Index 3 was the failed one.
Index 1 was needing recovery.

In this case, fault nodes = 2, which is greater than `F`.

Reply:
Yes, perhaps so.
Let's remove this NewViewNumber and make it static and see how the logic will be.

Reply:
Maybe it is safe to be moved back to `F` now that we are accepting any `(NewViewNumber >= viewNumber)`, what do you think, @erikzhang?

Reply:
Yes, let's move it back.

Reply:
`F+1` ? :dancer: 

Maybe it will speed-up a little bit in some cases.

Reply:
```cs
                    // Limit recovery to be sent from, at least, `f` nodes when the request is from a lower view number.
                    // However, `f+1` was chosen for ensuring a possible faster reply for those asking recovery
                    int allowedRecoveryNodeCount = context.F() + 1;
```

Reply:
`F` is okay.

Reply:
After we finish all this journey until Mainnet we can start to investigate things further.

![image](https://user-images.githubusercontent.com/17395762/55229108-d17fd380-51fa-11e9-89e6-bd345e1f8ba6.png)

Maybe an adjustment like `F+1` would reduce outliers like that.
But I agree with you, @erikzhang. It makes sense to use `F` now (It will also give us more insights about the behavior).